client/src/components/categorical/index.tsx

```diff
@@ -15,52 +15,32 @@ import { ControlsHelpers } from "../../util/stateManager";
 import { track } from "../../analytics";
 import { EVENTS } from "../../analytics/events";
 import { CATEGORICAL_SECTION_TEST_ID } from "./constants";
+import { AppDispatch, RootState } from "../../reducers";
+import { toggleCategoryExpansion } from "../../actions/controls";
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-type State = any;
-
-// @ts-expect-error ts-migrate(1238) FIXME: Unable to resolve signature of class decorator whe... Remove this comment to see the full error message
-@connect((state) => ({
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  schema: (state as any).annoMatrix?.schema,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  isCellGuideCxg: (state as any).controls.isCellGuideCxg,
-}))
-// eslint-disable-next-line @typescript-eslint/ban-types --- FIXME: disabled temporarily on migrate to TS.
-class Categories extends React.Component<{}, State> {
-  // eslint-disable-next-line @typescript-eslint/ban-types --- FIXME: disabled temporarily on migrate to TS.
-  constructor(props: {}) {
-    super(props);
-    this.state = {
-      expandedCats: new Set(),
```

Comment:
this is now a Redux state since thunk analytics actions need it. Also array is more Redux friendly than Set, since array is easier to copy and serialize without extra steps

Reply:
Array is slightly slower for larger arrays and Set is more performant, but I think for our array size that should be negligible?